src/tbb/arena.h

```diff
@@ -288,6 +289,8 @@ struct arena_base : padded<intrusive_list_node> {
     //! Used to trap accesses to the object after its destruction.
     std::uintptr_t my_guard;
 #endif /* TBB_USE_ASSERT */
+    //! Map from numa nodes to range of slots to this numa node
+    std::unordered_map<numa_node_id, std::pair<int, int>> numa_intervals;
```

Comment:
I would suggest add __TBB_HIERARCHICAL_STEALING_OPTIMIZATION and use inside TBB.
To decrease debug cost. (because class layouts are changed and so on)